src/responder/common/responder_common.c

```diff
@@ -108,9 +108,9 @@ static errno_t get_client_cred(struct cli_ctx *cctx)
     char cmd_line[255] = { 0 };
     int proc_fd;
 
-    cctx->creds->ucred.uid = -1;
     cctx->creds->ucred.gid = -1;
-    cctx->creds->ucred.pid = -1;
+    cli_creds_get_uid(cctx->creds) = -1;
+    cli_creds_get_pid(cctx->creds) = -1;
```

Comment:
I feel uneasy about using what _looks_ like an 'rvalue' as 'lvalue'.
(while `cli_creds_get_*()` are defines they look like a function)

Reply:
I propose to remove those lines at all.
Instead, add `talloc_zfree(cctx->creds)` below if `getsockopt()` fails.
Next patch won't be needed then.